src/js/_enqueues/admin/site-icon.js

```diff
@@ -1,138 +1,180 @@
-(function($) {
+/* global wp, jQuery */
+
+/**
+ * Calculates image selection options based on the attachment dimensions.
+ *
+ * @since 6.5.0
+ *
+ * @param {Object} attachment The attachment object representing the image.
+ * @return {Object} The image selection options.
+ */
+function calculateImageSelectOptions( attachment ) {
+	var realWidth = attachment.get( 'width' ),
+		realHeight = attachment.get( 'height' ),
+		xInit = 512,
+		yInit = 512,
+		ratio = xInit / yInit,
+		xImg = xInit,
+		yImg = yInit,
+		x1,
+		y1,
+		imgSelectOptions;
+
+	if ( realWidth / realHeight > ratio ) {
+		yInit = realHeight;
+		xInit = yInit * ratio;
+	} else {
+		xInit = realWidth;
+		yInit = xInit / ratio;
+	}
+
+	x1 = ( realWidth - xInit ) / 2;
+	y1 = ( realHeight - yInit ) / 2;
+
+	imgSelectOptions = {
+		aspectRatio: xInit + ':' + yInit,
+		handles: true,
+		keys: true,
+		instance: true,
+		persistent: true,
+		imageWidth: realWidth,
+		imageHeight: realHeight,
+		minWidth: xImg > xInit ? xInit : xImg,
+		minHeight: yImg > yInit ? yInit : yImg,
+		x1: x1,
+		y1: y1,
+		x2: xInit + x1,
+		y2: yInit + y1
+	};
+
+	return imgSelectOptions;
+}
+
+/**
+ * Initializes the media frame for selecting or cropping an image.
+ *
+ * @since 6.5.0
+ *
+ * @param {jQuery} $ The jQuery object.
+ */
+( function( $ ) {
 	var frame;
+	// Build the choose from library frame.
+	$( '#choose-from-library-link' ).on( 'click', function() {
+		var $el = $(this);
+
+		// Create the media frame.
+		frame = wp.media({
+			button: {
+				// Set the text of the button.
+				text: $el.data( 'update' ),
+				// Don't close, we might need to crop.
+				close: false
+			},
+			states: [
+				new wp.media.controller.Library({
+					title: $el.data( 'choose' ),
```

Comment:
This could be changed to 'choose-text' to avoid to add an unnecessary string on the PHP side.